polylabel.js

```diff
@@ -65,10 +73,10 @@ function polylabel(polygon, precision, debug) {
 
```

Comment:
```suggestion
        }
        
        // do not drill down further if there's no chance of a better solution
        if ((cell.max - cell.distanceToCentroid * centroidWeight) - fitness(bestCell) <= precision) continue;

```
Here's my suggestion to fix a potentially code-breaking issue of this line.
Given that `cell.max` represents the best possible result from a new cell to be compared, it should be equally adjusted as in the `fitness(cell)` function (subtracting the weighted `cell.distanceToCentroid` from the result to obtain the final "fitness".

Not making this change might result in an infinite loop, typically when the` centroidWeight `is much larger than the `precision`. e.g. `polylabel(polygon, 0.000001, false, 0.01);`